### PR TITLE
Unit test to create all repo test files against ach server

### PR DESCRIPTION
**Pull request by bkmoovio**
#412

### CODE REVIEW

**Comment by bkmoovio** [comment]
@adamdecaf ,  Please see if this pull request is more in line with what you are looking for.

File: iatBatch.go [hunk 1]
```diff
@@ -45,6 +46,27 @@ func NewIATBatch(bh *IATBatchHeader) IATBatch {
 	return iatBatch
 }
 
+// UnmarshalJSON un-marshals JSOn IATBatch
+func (batch *IATBatch) UnmarshalJSON(p []byte) error {
+	batch.Header = NewIATBatchHeader()
+	batch.Control = NewBatchControl()
+
+	if batch == nil {
```
Comment (adamdecaf) [inline iatBatch.go:54]: Won't the two lines above this panic if `batch == nil` ?

File: server/test/createJSONFromACHFiles_test.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+	}
+}
+
+type fileNames struct {
+	SECCode     string
+	AchFileName string
+	jsonName    string
+}
+
+func getFileNames() []fileNames {
```
Comment (adamdecaf) [inline server/test/createJSONFromACHFiles_test.go:100]: FYI, there's `path/filepath.Glob` that wouldn't require we list every file manually.

https://golang.org/pkg/path/filepath/#Glob

File: server/test/createJSONFromACHFiles_test.go [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+		if err != nil {
+			log.Fatal(err)
+		}
+		r := ach.NewReader(f)
+		achFile, err := r.Read()
+		if err != nil {
+			fmt.Printf("Issue reading file: %+v \n", err)
+		}
+		// ensure we have a validated file structure
+		if achFile.Validate(); err != nil {
```
Comment (adamdecaf) [inline server/test/createJSONFromACHFiles_test.go:34]:
```suggestion
		if err := achFile.Validate(); err != nil {
			t.Errorf("Could not validate entire read file: %v", err)
```

File: server/test/createJSONFromACHFiles_test.go [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+			fmt.Printf("Could not validate entire read file: %v", err)
+		}
+		// If you trust the file but it's formatting is off building will probably resolve the malformed file.
+		if achFile.Create(); err != nil {
```
Comment (adamdecaf) [inline server/test/createJSONFromACHFiles_test.go:38]:
```suggestion
		if err := achFile.Create(); err != nil {
			t.Errorf("Could not build file with read properties: %v", err)
```

File: server/test/createJSONFromACHFiles_test.go [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+			fmt.Printf("Could not build file with read properties: %v", err)
+		}
+
+		f.Close()
```
Comment (adamdecaf) [inline server/test/createJSONFromACHFiles_test.go:42]:
```suggestion
		if err := f.Close(); err != nil {
				t.Errof("unable to close %s: %v", file, err)
		}
```

File: server/test/createJSONFromACHFiles_test.go [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+
+	for _, file := range files {
+		f, err := os.Open(file.AchFileName)
+		if err != nil {
+			log.Fatal(err)
+		}
+		r := ach.NewReader(f)
+		achFile, err := r.Read()
+		if err != nil {
+			fmt.Printf("Issue reading file: %+v \n", err)
```
Comment (adamdecaf) [inline server/test/createJSONFromACHFiles_test.go:31]:
```suggestion
			t.Errorf("Issue reading file: %+v \n", err)
```

File: server/test/createJSONFromACHFiles_test.go [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+)
+
+// TestCreateJSONFromACHFiles creates JSON from existing ACH Files
+func TestCreateJSONFromACHFiles(t *testing.T) {
+	files := getFileNames()
+
+	for _, file := range files {
+		f, err := os.Open(file.AchFileName)
+		if err != nil {
+			log.Fatal(err)
```
Comment (adamdecaf) [inline server/test/createJSONFromACHFiles_test.go:26]:
```suggestion
			t.Error(err)
```

File: server/test/createJSONFromACHFiles_test.go [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+			}
+
+		}
+
+		bs, err := json.MarshalIndent(achFile, "", " ")
+		if err != nil {
+			fmt.Println("error:", err)
+		}
+
+		fmt.Printf("JSON Output: %v \r\n", string(bs))
```
Comment (adamdecaf) [inline server/test/createJSONFromACHFiles_test.go:61]: Do we need to print the JSON to stdout? That's going to be really noisy with `go test -v`.

File: server/test/createJSONFromACHFiles_test.go [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+
+		// Create ACH File from JSON
+		file, err := ach.FileFromJSON(bs)
+		if err != nil {
+			t.Fatal(err.Error())
+		}
+
+		// Validate ACH File
+		if file.Validate(); err != nil {
+			fmt.Printf("Could not validate file: %v", err)
```
Comment (adamdecaf) [inline server/test/createJSONFromACHFiles_test.go:73]:
```suggestion
			t.Errorf("Could not validate file: %v", err)
```

**Review by adamdecaf — CHANGES_REQUESTED** [review]
This looks great so far!

File: server/test/createJSONFromACHFiles_test.go (hunk 2 above)
Comment (bkmoovio) [reply server/test/createJSONFromACHFiles_test.go:100]: I know, but I wanted to control that.  It seemed risky to just do a pattern match.  If you want  a pattern match what would you want it on?

Comment (adamdecaf) [reply server/test/createJSONFromACHFiles_test.go:100]: Let's go with `../../test/ach-*-read/*.ach`.

File: iatBatch.go (hunk 1 above)
Comment (bkmoovio) [reply iatBatch.go:54]: I have not been able to recreate a panic when batch == nil, but I may not be creating it in a way you are thinking.

**Pull request merged**